src/main/java/saitama/parser/Parser.java

```diff
@@ -0,0 +1,192 @@
+package saitama.parser;
+
+import saitama.commands.AddCommand;
+import saitama.commands.Command;
+import saitama.commands.DeleteCommand;
+import saitama.commands.ExitCommand;
+import saitama.commands.FindCommand;
+import saitama.commands.ListCommand;
+import saitama.commands.MarkCommand;
+import saitama.commands.UnmarkCommand;
+import saitama.exceptions.EmptyDescriptionException;
+import saitama.exceptions.InvalidCommandException;
+import saitama.exceptions.InvalidFormatException;
+import saitama.exceptions.InvalidTaskNumberException;
+import saitama.exceptions.MissingQueryException;
+import saitama.tasks.Deadline;
+import saitama.tasks.Event;
+import saitama.tasks.Task;
+import saitama.tasks.ToDo;
+
+/**
+ * Interprets the user input.
+ */
+public class Parser {
+    /**
+     * Returns a Command object corresponding to the input given by the user.
+     *
+     * @param fullCommand The command input given by the user.
+     * @return Command object corresponding to the user input.
+     * @throws InvalidFormatException if command exists but is in invalid format.
+     * @throws EmptyDescriptionException if command exists but no details are given.
+     * @throws InvalidCommandException if command does not exist.
+     * @throws InvalidTaskNumberException if given command takes in a task number but the number does not exist.
+     */
+    public static Command parse(String fullCommand) throws InvalidFormatException, EmptyDescriptionException,
+            InvalidCommandException, InvalidTaskNumberException, MissingQueryException {
+        String[] splitCommand = fullCommand.split(" ", 2);
+        splitCommand[0] = splitCommand[0].toUpperCase();
+        String command = splitCommand[0];
+
+        switch (command) {
+        case "BYE":
+            return new ExitCommand();
+        case "LIST":
+            return new ListCommand();
+        case "MARK":
+            return prepareMark(splitCommand);
+        case "UNMARK":
+            return prepareUnmark(splitCommand);
+        case "DELETE":
+            return prepareDelete(splitCommand);
+        case "FIND":
+            return prepareFind(splitCommand);
+        case "TODO":
+            //Fallthrough
+        case "DEADLINE":
+            //Fallthrough
+        case "EVENT":
+            //Fallthrough
+            return prepareAdd(splitCommand);
+        default:
+            throw new InvalidCommandException();
```

Comment:
Agree with Seth! Really liked the idea of abstracting every single command into its own function in its respective class. As a reader I feel comfortable in reading your code.